### PR TITLE
Add marker example for javascript

### PR DESCRIPTION
**Pull request by gpollo**
This simple example computes the region of 5 ms containing the most
`sched_switch` events and highlights it using a marker.

Signed-off-by: Gabriel-Andrew Pollo-Guilbert <gabrielpolloguilbert@gmail.com>

### CODE REVIEW

**Comment by tahini** [comment]
Could you try to make the python script run with jython instead of py4j? Jython is much more performant for our purposes. As proof, see http://versatic.net/tracecompass/scriptEngineBenchmarks.html

File: examples/python/marker.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+from py4j.java_gateway import JavaClass
+from py4j.protocol import Py4JError
+
+loadModule('/TraceCompass/Analysis')
+loadModule('/TraceCompass/DataProvider')
+loadModule('/TraceCompass/View')
+loadModule('/TraceCompass/Trace')
+loadModule('/TraceCompass/TraceMarker')
+
+analysis = getActiveTrace()
```
Comment (tahini) [inline examples/python/marker.py]: trace =

File: examples/python/marker.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        if event.getName() == "sched_switch":
+            updateStateSystem(event)
+            
+            # TODO: Py4J throws exceptions sometimes, we ignore them for the time being
+            #       event though it causes the Python version to give different results
+            #       from the Javascript version
+            try:
+                updateMostEventful(event, duration)
+            except Py4JError:
+                pass
```
Comment (tahini) [inline examples/python/marker.py]: This try catch hides an exception in the trace. Removing this, I get the following exception:

org.eclipse.ease.ScriptExecutionException: Traceback (most recent call last):
  File "workspace://test_traces/marker.py", line 98, in <module>
  File "workspace://test_traces/marker.py", line 89, in runAnalysis
  File "workspace://test_traces/marker.py", line 60, in updateMostEventful
  File "/home/gen/workspace/Dorsal/tcincu-workspace/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/py4j-python_0.10.8.1-bnd-2odeag/src/py4j/java_gateway.py", line 1286, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/home/gen/workspace/Dorsal/tcincu-workspace/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/py4j-python_0.10.8.1-bnd-2odeag/src/py4j/protocol.py", line 332, in get_return_value
    format(target_id, ".", name, value))
Py4JError: An error occurred while calling o34.getTimestamp. Trace:
py4j.Py4JException: Target Object ID does not exist for this gateway :o34
	at py4j.Gateway.invoke(Gateway.java:279)

And the marker that I get is the first one only, ie not the same as the javascript marker

**Review by tahini — CHANGES_REQUESTED** [review]
Thanks for this, javascript works great. Python, not as great. You can either fix the python script or separate javascript and python PR and I can merge the javascript one and eventually debug the python one.

**Comment by gpollo** [comment]
Alright, I'll put the python example in a separate PR.

**Comment by tahini** [comment]
Thanks!

**Pull request merged**